csi/node_server.go

```diff
@@ -852,3 +840,31 @@ func (ns *NodeServer) getMounter(volume *longhornclient.Volume, volumeCapability
 
 	return nil, fmt.Errorf("failed to get mounter for volume %v unsupported volume capability %v", volume.Name, volumeCapability.GetAccessType())
 }
+
+// restageRequired determines whether it is necessary to manually call NodeUnstageVolume and NodeStageVolume again
+// before publishing. If it returns true, it may also return an error containing the underlying reason restaging is
+// required. restageRequired has side effects for v1 mount volumes due to its use of ensureMountPoint. These side
+// effects are neither harmful nor helpful, as ensureMountPoint will be called again in the restage flow.
+func restageRequired(volume *longhornclient.Volume,
+	volumeID, stagingTargetPath string,
+	mounter mount.Interface,
+	isBlock bool) (bool, error) {
+
+	if volume.DataEngine == string(longhorn.DataEngineTypeV2) {
+		return true, fmt.Errorf("always unstage v2 volume %v", volumeID)
+	}
```

Comment:
Could you elaborate more on the v2 volume check? Thank you.

Reply:
This logic is copied directly from lines 173-174 before the changes. I added the `true` return because of the way the function works now, but the previous logic made `unstageRequired` true for v2 volumes and logged this exact message. TBH, I don't know exactly why we need(ed) this behavior, but I don't think this PR should change it.

Reply:
If not required, we can remove it. Derek?